ipatests/test_xmlrpc/test_permission_plugin.py

```diff
@@ -4345,3 +4345,48 @@ class test_autoadd_operational_attrs(Declarative):
             'allow (read) groupdn = "ldap:///%s";)' % permission1_dn,
         ),
     ]
+
+
+class test_self_bindrule(Declarative):
+    """Test creation of permission with bindrule self
+    """
+    cleanup_commands = [
+        ('permission_del', [permission1], {'force': True}),
+    ]
+
+    tests = [
+        dict(
+            desc='Create %r' % permission1,
+            command=(
+                'permission_add', [permission1], dict(
+                    ipapermlocation=DN('cn=accounts', api.env.basedn),
+                    ipapermright=u'read',
+                    ipapermbindruletype='self',
+                    attrs=[u'ObjectClass'],
+                )
+            ),
+            expected=dict(
+                value=permission1,
+                summary=u'Added permission "%s"' % permission1,
+                result=dict(
+                    dn=permission1_dn,
+                    cn=[permission1],
+                    objectclass=objectclasses.permission,
+                    attrs=[u'ObjectClass', u'entryusn', u'createtimestamp',
+                           u'modifytimestamp'],
+                    ipapermright=[u'read'],
+                    ipapermbindruletype=[u'self'],
+                    ipapermissiontype=[u'SYSTEM', u'V2'],
+                    ipapermlocation=[DN('cn=accounts', api.env.basedn)],
+                ),
+            ),
+        ),
+
+        verify_permission_aci(
+            permission1, DN('cn=accounts', api.env.basedn),
+            '(targetattr = "ObjectClass || createtimestamp || entryusn || '
+            + 'modifytimestamp")'
+            + '(version 3.0;acl "permission:%s";' % permission1
+            + 'allow (read) userdn = "ldap:///self";)',
+        ),
+    ]
```

Comment:
A bit of a nit but we try to keep attributes lower-case in the python code and ObjectClass really stands out in any case.

Reply:
Can you expand in the commit message that this is so a plugin can provide its own self-service rule and not rely on the global one? This is quite useful in particular for the external use case you found in the FAS plugin.

Reply:
Good point, I copy pasta the code from another test and didn't bother to change any non-related strings.